linux/ibus-keyman/src/engine.c

```diff
@@ -40,11 +40,18 @@
 #include <keyman/keyboardprocessor.h>
 #include <keyman/keyboardprocessor_consts.h>
 
+#include "config.h"
 #include "keymanutil.h"
 #include "keyman-service.h"
 #include "engine.h"
 #include "keycodes.h"
 
+// Fallback for older ibus versions that don't define IBUS_PREFILTER_MASK
+#ifndef IBUS_HAS_PREFILTER
+#warning Updated ibus version is not installed. Ordered output will not be available.
```

Comment:
The warning as it stands is not very helpful to a dev encountering it. Perhaps we should reference a specific PR in the warning?

```suggestion
#warning Compiling against ibus version that does not include prefilter mask patch (https://github.com/ibus/ibus/pull/2440). Output ordering guarantees will be disabled.
```